### PR TITLE
add and style the new metrics chart tooltip

### PR DESCRIPTION
**Pull request by jacobheric**
## Description

Adds a custom tooltip component that better matches the designs and shims it into the charts. 
Screenshot 2024-10-09 at 4 48 16 PM
Screenshot 2024-10-09 at 4 48 08 PM

## Motivation
Better tooltip

## Type of change (choose one)
- [x] Chore (refactors, upgrades, etc.)
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Security fix (non-breaking change that fixes a potential vulnerability)
- [ ] Docs
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

## Checklist
- [ ] I've linked any associated issues to this PR.
- [x] I've tested my own changes.

*[Check our Pull Request Guidelines](https://github.com/inngest/inngest/blob/main/docs/PULL_REQUEST_GUIDELINES.md)*

### CODE REVIEW

File: ui/apps/dashboard/src/components/Metrics/utils.ts [hunk 1]
```diff
@@ -93,9 +99,16 @@ export const getLineChartOptions = (
...
       appendTo: () => document.getElementById('chart-tooltip'),
-      extraCssText: 'max-height: 350px; overflow-y: scroll;',
-      className: 'no-scrollbar',
-      transitionDuration: 1.5,
+      transitionDuration: 1,
+      //
+      // rendering our tooltip componen to a string doesn't support click events
+      // so wrap the whole thing in a vanilla js clipboard cliip handler
+      formatter: (params: any) =>
+        `<div onclick="navigator.clipboard.writeText('${
```
Comment (jacobheric) [inline ui/apps/dashboard/src/components/Metrics/utils.ts:107]: A bit of shenanigans here. I'm rendering the thing to a string so we can use an actual component with all the tailwind styles and icon trimmings. But we can send events along that way so I'm just wrapping it up in a vanilla js event handler.

File: ui/apps/dashboard/src/components/Metrics/ChartTooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import type { DefaultLabelFormatterCallbackParams } from '@inngest/components/Chart/Chart.jsx';
```
Comment (anafilipadealmeida) [inline ui/apps/dashboard/src/components/Metrics/ChartTooltip.tsx:1]: You probably need to remove the .jsx from that import, I'm assuming that happened thanks to the autocomplete

File: ui/apps/dashboard/src/components/Metrics/utils.ts [hunk 3]
```diff
@@ -93,9 +99,16 @@ export const getLineChartOptions = (
       // Attach tooltips to a dedicated dom node above interim parents
       // with low z-indexes
       appendTo: () => document.getElementById('chart-tooltip'),
-      extraCssText: 'max-height: 350px; overflow-y: scroll;',
-      className: 'no-scrollbar',
-      transitionDuration: 1.5,
+      transitionDuration: 1,
+      //
+      // rendering our tooltip componen to a string doesn't support click events
```
Comment (anafilipadealmeida) [inline ui/apps/dashboard/src/components/Metrics/utils.ts:104]: Nitpicking: some typos in the comment

**Review by anafilipadealmeida — APPROVED** [review]
Approving this already, but I think we need to fix one import (left a comment there)

File: ui/apps/dashboard/src/components/Metrics/ChartTooltip.tsx (hunk 2 above)
Comment (jacobheric) [reply ui/apps/dashboard/src/components/Metrics/ChartTooltip.tsx]: Nice catch, fixed.

File: ui/apps/dashboard/src/components/Metrics/utils.ts (hunk 3 above)
Comment (jacobheric) [reply ui/apps/dashboard/src/components/Metrics/utils.ts]: fixed

**Pull request merged**